src/pages/editor/curators/AddCuratorDialog.tsx

```diff
@@ -0,0 +1,180 @@
+import { useDebounce } from '../../../utils/hooks/useDebounce';
+import { getAllChildOrganizations, getOrganizationHierarchy } from '../../../utils/institutions-helpers';
+import { getFullCristinName, getUsername } from '../../../utils/user-helpers';
+import { AddCuratorForm } from './AddCuratorForm';
+import { OrganizationCuratorsAccordionProps } from './OrganizationCuratorsAccordion';
+
+interface AddCuratorDialogProps
+  extends Pick<DialogProps, 'open'>,
+    Pick<OrganizationCuratorsAccordionProps, 'refetchCurators'> {
+  onClose: () => void;
+  currentOrganization: Organization;
+}
+
+export const AddCuratorDialog = ({ onClose, open, currentOrganization, refetchCurators }: AddCuratorDialogProps) => {
+  const { t } = useTranslation();
+  const user = useSelector((store: RootState) => store.user);
+  const topOrgCristinId = user?.topOrgCristinId ?? '';
+
+  const [searchQuery, setSearchQuery] = useState('');
+  const debouncedSearchQuery = useDebounce(searchQuery);
+
+  const [selectedPerson, setSelectedPerson] = useState<CristinPerson | null>(null);
+  const [userInitialValues, setUserInitialValues] = useState<InstitutionUser | null>(null);
+
+  const closeDialog = () => {
+    setSearchQuery('');
+    setSelectedPerson(null);
+    setUserInitialValues(null);
+    onClose();
+  };
+
+  const employeeSearchQuery = useQuery({
+    enabled: open && !!topOrgCristinId && !!debouncedSearchQuery && debouncedSearchQuery === searchQuery,
+    queryKey: ['employees', topOrgCristinId, 20, 1, debouncedSearchQuery],
+    queryFn: ({ signal }) => fetchEmployees(topOrgCristinId, 20, 1, debouncedSearchQuery, signal),
+    meta: { errorMessage: t('feedback.error.get_users_for_institution') },
+  });
+
+  const username = getUsername(selectedPerson, topOrgCristinId);
+
+  const userQuery = useQuery({
+    enabled: open && !!selectedPerson && !!username,
+    queryKey: ['user', username],
+    queryFn: () => fetchUser(username),
+    meta: { errorMessage: false }, // No error message, since a Cristin Person will lack User if they have not logged in yet
+    retry: false,
+    cacheTime: 0, // Disable caching since this user in many cases will be changed, and the cached data then will be outdated
+  });
+
+  useEffect(() => {
+    const currentUser = userQuery.data;
+    if (currentUser) {
+      let viewingScope = currentUser.viewingScope.includedUnits;
+
+      const newOrganizationId = !viewingScope.includes(currentOrganization.id) ? currentOrganization.id : null;
+      if (newOrganizationId) {
+        // Remove organizations conflicting with the new organization
+        const parentOrganizationIds = getOrganizationHierarchy(currentOrganization)
+          .filter((org) => org.id !== currentOrganization.id)
+          .map((org) => org.id);
+        const childOrganizationIds = getAllChildOrganizations(currentOrganization.hasPart).map((unit) => unit.id);
+        viewingScope = viewingScope.filter(
```

Comment:
ugh, fant en liten bug for store institusjoner med flere nivå her, da parent ikke fjernes om det er mer enn ett nivå over.